File: app/src/main/java/com/upsage/evosummermobilelab/data/converters/DateConverter.java
package com.upsage.evosummermobilelab.data.converters;

import androidx.room.TypeConverter;

import java.util.Date;

public class DateConverter {
    @TypeConverter
    public Long dateToTimestamp(Date date) {
        if (date == null) {
            return null;
        } else {
            return date.getTime();
        }
    }

    @TypeConverter
    public Date timestampToDate(long time) {
        return new Date(time);
    }
}

File: app/src/main/java/com/upsage/evosummermobilelab/data/DateConverter.kt
package com.upsage.evosummermobilelab.data

import androidx.room.TypeConverter
import java.util.*

class DateConverter {
    @TypeConverter
    fun dateToTimestamp(date: Date?): Long? {
        return date?.time
    }

    @TypeConverter
    fun timestampToDate(time: Long): Date {
        return Date(time)
    }
}
